src/fmu/dataio/dataio.py

```diff
@@ -812,47 +812,31 @@ class InitializeCase:  # pylint: disable=too-few-public-methods
             some predefined main level keys. If config is None or the env variable
             FMU_GLOBAL_CONFIG pointing to a file is provided, then it will attempt to
             parse that file instead.
-        rootfolder: To override the automatic and actual ``rootpath``. Absolute path to
-            the case root, including case name. If not provided (which is not
-            recommended), the rootpath will be attempted parsed from the file structure
-            or by other means.
+        rootfolder: Absolute path to the case root, including case name.
         casename: Name of case (experiment)
         caseuser: Username provided
-        restart_from: ID of eventual restart (deprecated)
-        description: Description text as string or list of strings.
+        description (Optional): Description text as string or list of strings.
     """
 
     # class variables
     meta_format: ClassVar[Literal["yaml", "json"]] = "yaml"
 
-    # instance
     config: dict
-    rootfolder: Optional[Union[str, Path]] = None
-    casename: Optional[str] = None
-    caseuser: Optional[str] = None
-    restart_from: Optional[str] = None
+    rootfolder: str | Path
+    casename: str
+    caseuser: str
+
     description: Optional[Union[str, list]] = None
-    verbosity: str = "DEPRECATED"
 
     _metadata: dict = field(default_factory=dict, init=False)
     _metafile: Path = field(default_factory=Path, init=False)
     _pwd: Path = field(default_factory=Path, init=False)
     _casepath: Path = field(default_factory=Path, init=False)
 
     def __post_init__(self) -> None:
-        if self.verbosity != "DEPRECATED":
-            warn(
-                "Using the 'verbosity' key is now deprecated and will have no "
-                "effect and will be removed in near future. Please remove it from the "
-                "argument list. Set logging level from client script in the standard "
-                "manner instead.",
-                UserWarning,
-            )
```

Comment:
This would only be passed from create_case_metadata.py, which did not pass it, so doesn't appear to me relevant to keep this.